components/cameraScreen.js

```diff
@@ -0,0 +1,199 @@
+  const isFocused = useIsFocused();
+
+  React.useLayoutEffect(() => {
+    navigation.setOptions({ tabBarVisible: false });
+  }, [navigation]);
+
+  useEffect(() => {
+    (async () => {
+      const { status } = await Camera.requestPermissionsAsync();
+      setHasPermission(status === "granted");
+    })();
+  }, []);
+  takePicture = async () => {
+    if (this.camera) {
+      await this.camera.takePictureAsync({
+        onPictureSaved: this.onPictureSaved,
+      });
+    }
+  };
+  const windowWidth = Dimensions.get("window").width;
+  const windowHeight = Dimensions.get("window").height;
+  console.log(windowHeight / windowWidth);
+
+  function makeid(length) {
+    var result = "";
+    var characters =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    var charactersLength = characters.length;
+    for (var i = 0; i < length; i++) {
+      result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    }
+    return result;
+  }
+  onPictureSaved = async (photo) => {
+    console.log(photo);
+
+    var imageList = [];
+    var existingImages = false;
+    try {
+      const value = await AsyncStorage.getItem("@Images");
+      console.log("this is value: ");
+      console.log(value);
+      console.log(value == "null");
+      const valueObject = JSON.parse(value);
+      // console.log(valueObject.length )
+      if (valueObject && valueObject.length != 0) {
+        existingImages = true;
+        for (var i = 0; i < valueObject.length; i++) {
+          imageList.push(valueObject[i]);
+        }
+      }
+      imageList.unshift({ uri: photo.uri, id: makeid(8) });
+      await AsyncStorage.setItem("@Images", JSON.stringify(imageList));
+    } catch (error) {
+      // Error saving data
+      console.log(error);
+    }
+
+    // console.log(result);
+    navigation.reset({
+      index: 0,
+      routes: [{ name: "PicturePage" }],
+    });
+
+    //   console.log(result);
+  };
+
+  if (hasPermission === null) {
+    return <View />;
+  } else if (hasPermission === false) {
+    return <Text>No access to camera</Text>;
+  }
+  return (
+    <View style={styles.view}>
+      {isFocused && (
+        <Camera
+          ratio="2:1"
+          style={{ width: windowWidth, height: windowWidth * 2 }}
+          type={type}
+          ref={(ref) => {
+            this.camera = ref;
+          }}
+        ></Camera>
+      )}
+      <View
+        style={{
+          position: "absolute",
+          top: 0,
+          left: 0,
+          right: 0,
+          bottom: 0,
+          justifyContent: "flex-end",
+          alignItems: "center",
+          paddingBottom: 30,
+        }}
+      >
+        <TouchableOpacity
+          onPress={() => {
+            takePicture();
+          }}
+        >
+          <SimpleLineIcons name="camera" size={42} color="white" />
+        </TouchableOpacity>
+      </View>
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
+  view: {
```

Comment:
Some of these styles aren't used in this component (ex. `viewCenter`, `name`). Please clean this up